test/e2e_node/split_disk_test.go

```diff
@@ -0,0 +1,338 @@
+
+// InodeEviction tests that the node responds to node disk pressure by evicting only responsible pods.
+// Node disk pressure is induced by consuming all inodes on the node.
+
+// Existing Eviction tests are quite flaky and need some investigation
+// Flakiness arises from ordering
+// Stats is best effort and we evict based on stats being successful
+
+// Container runtime filesystem should display different stats for imagefs and nodefs
+
+var _ = SIGDescribe("Summary with split filesystem", framework.WithSlow(), framework.WithSerial(), framework.WithDisruptive(), nodefeature.SplitDisk, func() {
+	f := framework.NewDefaultFramework("summary-test")
+	f.It("should display different stats for containerfs and nodefs", func(ctx context.Context) {
+		summary := eventuallyGetSummary(ctx)
+		gomega.Expect(summary.Node.Fs).ToNot(gomega.Equal(summary.Node.Runtime.ImageFs))
+		gomega.Expect(summary.Node.Fs.AvailableBytes).To(gomega.Equal(summary.Node.Runtime.ContainerFs.AvailableBytes))
+		gomega.Expect(summary.Node.Fs.CapacityBytes).To(gomega.Equal(summary.Node.Runtime.ContainerFs.CapacityBytes))
+
+	})
+})
+
+// DiskPressureImageFs tests that the node responds to node disk pressure by evicting only responsible pods
+// Disk pressure is induced by running dd on imagefs.
+var _ = SIGDescribe("DiskPressureImageFs", framework.WithSlow(), framework.WithSerial(), framework.WithDisruptive(), nodefeature.SplitDisk, func() {
+	ginkgo.BeforeEach(func(ctx context.Context) {
+		e2eskipper.SkipUnlessFeatureGateEnabled(features.KubeletSeparateDiskGC)
+		if !hasSplitFileSystem(ctx) {
+			ginkgo.Skip("it doesn't have split filesystem")
+		}
+	})
+
+	f := framework.NewDefaultFramework("diskpressure-imagefs-test")
+	f.NamespacePodSecurityLevel = admissionapi.LevelPrivileged
+	pressureTimeout := 10 * time.Minute
+	expectedNodeCondition := v1.NodeDiskPressure
+	ginkgo.Context(fmt.Sprintf(testContextFmt, expectedNodeCondition), func() {
+		tempSetCurrentKubeletConfig(f, func(ctx context.Context, initialConfig *kubeletconfig.KubeletConfiguration) {
+			initialConfig.EvictionHard = map[string]string{string(evictionapi.SignalNodeFsAvailable): "30%", string(evictionapi.SignalContainerFsAvailable): "30%", string(evictionapi.SignalImageFsAvailable): "30%"}
+			initialConfig.EvictionMinimumReclaim = map[string]string{}
+			ginkgo.By(fmt.Sprintf("EvictionHard %s", initialConfig.EvictionHard))
+
+		})
+
+		runImageFsPressureTest(f, pressureTimeout, expectedNodeCondition, logDiskMetrics, []podEvictSpec{
+			{
+				evictionPriority: 1,
+				pod:              innocentPod(),
+			},
+		})
+	})
+})
+
+// StorageVolumeEviction tests that the node responds to node disk pressure by evicting pods.
+// Volumes write to the node filesystem so we are testing eviction on nodefs even if it
+// exceeds imagefs limits.
+var _ = SIGDescribe("StorageVolumeEviction", framework.WithSlow(), framework.WithSerial(), framework.WithDisruptive(), nodefeature.SplitDisk, func() {
```

Comment:
since the SIGDescribe fields are largely the same (besides the title) can we compress all of these into one, and then have different ginkgo.Contexts for the different test names?

Reply:
Thanks! I grouped them.